crates/fluvio-connector/src/config.rs

```diff
@@ -0,0 +1,50 @@
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+use std::fs::File;
+use std::io::Read;
+
+use fluvio_controlplane_metadata::connector::ManagedConnectorSpec;
+use crate::error::ConnectorError;
+
+#[derive(Debug, Deserialize)]
+pub struct ConnectorConfig {
+    #[serde(default = "String::new")]
```

Comment:
is this necessary?

Reply:
It's annoying because if you wanna specify a yaml such as:
```yaml
foo:
  type: syslog
  args:
    arg1key: arg1value
```

where `foo` is the `name`, you have to parse the above as a `BTreeMap<String, ConnectorConfig>` or do something else.

Reply:
I think we should make attribute explicit (similar to K8 configuration) and addsome preamble such as 'version`. so config could be something like:
```
version: v1
type: test-connector
metadata:
     name: test
config:
   topic: my-log
   args:
      foo: bar
```

Reply:
After the discussion this seems like the way forward:
```yaml
---
version: v1
name: my-test-connector
type: test-connector
direction: source
args:
  foo: bar
secrets:
  foo: bar
```